### PR TITLE
Adds additional unsigned primitive support and test coverage

### PR DESCRIPTION
**Pull request by tosmun-r7**
- Uses appropriate return type for: `PacketInput.readUnsignedByte()`
- Adds convenience method: `PacketInput.readUnsignedInt()`
- Adds convenience method: `PacketOutput.writeInt(long)`
- Adds full unit test coverage including signed/unsigned parameters for `PacketInput`/`PacketOutput`
- Improves `RPCUnicodeString` and `RPCSID`:
  * Each of these structs now read/write unsigned primitives where required
  * Now skipping bytes instead of parsing numbers where the values are not used

### CODE REVIEW

File: src/main/java/com/rapid7/client/dcerpc/objects/RPCUnicodeString.java [hunk 1]
```diff
@@ -200,39 +196,33 @@ public void unmarshalDeferrals(PacketInput in) throws IOException {
...
 
             //Entity
             // <NDR: unsigned long> Offset for varying array
             // Alignment: 4 - Already aligned
-            final int offset = in.readInt();
+            final int offset = readIndex(in);
             // <NDR: unsigned long> ActualCount for varying array
             // Alignment: 4 - Already aligned
-            final int actualCount = in.readInt();
+            final int actualCount = readIndex(in);
```
Comment (sangdo-lee) [inline src/main/java/com/rapid7/client/dcerpc/objects/RPCUnicodeString.java]: Q. Did you find any case passing argument with singed int (i.e.negative value)?
Anyway, to remove signed int you might add the bit & operation(FFFF FFFF) for safety?

Comment (tosmun-r7) [reply src/main/java/com/rapid7/client/dcerpc/objects/RPCUnicodeString.java]: `readIndex` was implemented here: https://github.com/rapid7/smbj-rpc/pull/57/files#diff-8390af140f5b6f61873bcbec7f8cd8d2R255

It uses `PacketInput.readUnsignedInt()`, which will use `& 0xFFFFFFFF` as you are suggesting: https://github.com/rapid7/smbj-rpc/pull/57/files#diff-bbe35e255b3d87886cd9524f169d0291R99

We should never get a negative number in the response this way, as we never interpret it that way. :+1:

Comment (sangdo-lee) [reply src/main/java/com/rapid7/client/dcerpc/objects/RPCUnicodeString.java]: cool !

**Comment by sangdo-lee** [comment]
LGTM

**Pull request merged**